src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -279,7 +279,7 @@ private void EnsureValidRecorderStates()
         /// <param name="model">The model to check against.</param>
         /// <returns>Returns true if the model has already been recorded in the
         /// current action group, or false otherwise.</returns>
-        private bool IsRecordedInActionGroup(XmlElement group, ModelBase model)
+        private bool IsRecordedInActionGroup(XmlElement group, ModelBase model, UserAction action)
```

Comment:
The reason we did not have `UserAction` here is because it shouldn't matter. I can't say I remember what was the scenario, but for example if a code block node is created *and* edited in the same action group, the editing should not be recorded (otherwise the first undo will result in just undoing the edit, instead of removing the whole code block). This may not be the scenario but I'm afraid there might be others.

For your case, I was wondering why did we record the `ConnectorModel` creation in the first place if it is not meant to last. Can we not do the thing `ModelModificationUndoHelper` does, where it compares the connector set *before* and *after* the editing, then try to record only the real difference (whatever that comes up and goes away in the middle, does not get recorded). Does that make sense?